source/agora/network/Manager.d

```diff
@@ -1197,78 +1197,16 @@ public class NetworkManager
-
-        In the default implementation, this returns a `BlockExternalizedHandler`
-
-        Params:
-            address = The address (IPv4, IPv6, hostname) of target Server
-        Returns:
-            A BlockExternalizedHandler to communicate with the server
-            at `address`
-
-    ***************************************************************************/
-
-    public BlockExternalizedHandler getBlockExternalizedHandler
-        (Address address)
-    {
-        import vibe.http.client;
-
-        auto settings = new RestInterfaceSettings;
-        settings.baseURL = URL(address);
-        settings.httpClientSettings = new HTTPClientSettings;
-        settings.httpClientSettings.connectTimeout = this.node_config.timeout;
-        settings.httpClientSettings.readTimeout = this.node_config.timeout;
-        settings.httpClientSettings.proxyURL = this.proxy_url;
-
-        return new RestInterfaceClient!BlockExternalizedHandler(settings);
-    }
-
-    /***************************************************************************
-
-        Instantiates a client object implementing `PreImageReceivedHandler`
-
-        In the default implementation, this returns a `PreImageReceivedHandler`
-
-        Params:
-            address = The address (IPv4, IPv6, hostname) of target Server
-        Returns:
-            A PreImageReceivedHandler to communicate with the server
-            at `address`
-
-    ***************************************************************************/
-
-    public PreImageReceivedHandler getPreimageReceivedHandler
-        (Address address)
-    {
-        import vibe.http.client;
-
-        auto settings = new RestInterfaceSettings;
-        settings.baseURL = URL(address);
-        settings.httpClientSettings = new HTTPClientSettings;
-        settings.httpClientSettings.connectTimeout = this.node_config.timeout;
-        settings.httpClientSettings.readTimeout = this.node_config.timeout;
-        settings.httpClientSettings.proxyURL = this.proxy_url;
-
-        return new RestInterfaceClient!PreImageReceivedHandler(settings);
-    }
-
-    /***************************************************************************
-
-        Instantiates a client object implementing `TransactionReceivedHandler`
-
-        In the default implementation, this returns a `TransactionReceivedHandler`
+        Instantiates a client object implementing the templated `Handler`
 
         Params:
             address = The address (IPv4, IPv6, hostname) of target Server
         Returns:
-            A TransactionReceivedHandler to communicate with the server
-            at `address`
+            A Handler to communicate with the server at `address`
 
     ***************************************************************************/
 
-    public TransactionReceivedHandler getTransactionReceivedHandler
-        (Address address)
+    public Handler getHandler (Handler)(Address address)
```

Comment:
That's not possible. Templated function are not virtual.

Reply:
Reverted back to handler for each but extracted repetition into new function `getRestInterfaceSettings`